### PR TITLE
CO-2416_ActorApiUser_is_missing_from_Identifier_Assignment_via_Namespace_API_impelementation

### PR DESCRIPTION
**Pull request by ioigoume**
Add Actor Api User support and missing $db variable initialization in IdentifierAssignment Model

### CODE REVIEW

File: app/Model/CoIdentifierAssignment.php [hunk 1]
```diff
@@ -255,6 +255,7 @@ public function assign($coIdentifierAssignment, $objType, $objId, $actorCoPerson
     
     if($coIdentifierAssignment['CoIdentifierAssignment']['algorithm'] == IdentifierAssignmentEnum::Plugin) {
       if(empty($coIdentifierAssignment['CoIdentifierAssignment']['plugin'])) {
+        $dbc->commit();
```
Comment (boshrin) [inline app/Model/CoIdentifierAssignment.php:258]: I think these `commit`s should be `rollback`s?

File: app/Model/CoIdentifierAssignment.php [hunk 2]
```diff
@@ -499,7 +502,14 @@ private function checkInsert($coIdentifierAssignment, $objType, $assignEmail, $o
     
     $coId = $coIdentifierAssignment['CoIdentifierAssignment']['co_id'];
     $fk = Inflector::underscore($objType) . "_id";
-    
+
+
+    // Begin a transaction. This is more because we need to ensure the integrity of
+    // data between SELECT and INSERT/UPDATE than that we expect to rollback.
+
+    $dbc = $this->getDataSource();
```
Comment (boshrin) [inline app/Model/CoIdentifierAssignment.php:510]: Why are we starting a new transaction here? `checkInsert()` should be called from within a transaction.

Comment (ioigoume) [reply app/Model/CoIdentifierAssignment.php]: @boshrin thank you for the clarification. `checkInsert()` is using `$dbc->rollback()` but there is no initialization of the `$dbc`. I wrongly assumed that the function should handle the transaction by itself. Following your comment, i removed the transaction initialization and passed the `$dbc` variable as a parameter. Now `checkInsert()` has a reference to the transaction and can call a rollback if an exception occurs.

Comment (boshrin) [reply app/Model/CoIdentifierAssignment.php]: `checkInsert()` doesn't need to call the rollback. If it throws an Exception, the parent code will catch it and call `rollback()`.

Comment (ioigoume) [reply app/Model/CoIdentifierAssignment.php]: Done

**Pull request merged**